lib/arch/avx2/enc_loop.c

```diff
@@ -1,22 +1,27 @@
 // If we have AVX2 support, pick off 24 bytes at a time for as long as we can.
 // But because we read 32 bytes at a time, ensure we have enough room to do a
 // full 32-byte read without segfaulting:
-while (srclen >= 32)
-{
-	// Load string:
-	__m256i str = _mm256_loadu_si256((__m256i *)c);
 
-	// Reshuffle:
-	str = enc_reshuffle(str);
+if(srclen >= 32) {
+	const uint8_t* const o_orig = o;
```

Comment:
What is the use of this `o_orig`? Why not modify `o` directly? To keep the increment of `outl` out of the hot loop?